photon-ml/src/integTest/scala/com/linkedin/photon/ml/function/ObjectiveFunctionIntegTest.scala

```diff
@@ -16,402 +16,554 @@ package com.linkedin.photon.ml.function
 
 import java.util.Random
 
-import scala.collection.mutable
-
 import breeze.linalg.{DenseVector, SparseVector, Vector}
 import com.linkedin.photon.ml.data.LabeledPoint
-import com.linkedin.photon.ml.optimization._
+import com.linkedin.photon.ml.supervised.TaskType
 import com.linkedin.photon.ml.test.SparkTestUtils
 import org.apache.log4j.{LogManager, Logger}
+import org.apache.spark.rdd.RDD
 import org.testng.Assert.{assertEquals, assertTrue}
 import org.testng.annotations.{DataProvider, Test}
 
-
 /**
- * Unit tests to verify functions have gradients / Hessians to be consistent with each other.
- * @author bdrew
- * @author yali
- */
+  * Integration tests to verify that the loss functions compute gradients & Hessians accurately.
+  */
 class ObjectiveFunctionIntegTest extends SparkTestUtils {
```

Comment:
@joshvfleming 
Very similar changes to that of the ObjectiveFunctionTest unit tests class.